extensions/objects/templates/literally-canvas-diagram-editor.component.spec.ts

```diff
@@ -59,9 +134,264 @@ describe('LiterallyCanvasDiagramEditor', function() {
     expect(LCDiagramEditorCtrl.getDiagramSizeInfo()).toBe(helpText);
   });
 
-  it('should validate data', function() {
-    // TODO(#9357): Will be implemented once the svg data saving
-    // functionality is implemented.
-    expect(LCDiagramEditorCtrl.validate()).toBe(false);
+  it('should check if diagram is created', function() {
+    LCDiagramEditorCtrl.lc.currentSvg = defaultsvg;
+    expect(LCDiagramEditorCtrl.isDiagramCreated()).toBe(false);
+    LCDiagramEditorCtrl.lc.currentSvg = linesvg;
+    expect(LCDiagramEditorCtrl.isDiagramCreated()).toBe(true);
+  });
+
+  it('should check whether user is drawing', function() {
+    expect(LCDiagramEditorCtrl.isUserDrawing()).toBe(false);
+    LCDiagramEditorCtrl.lc._shapesInProgress = ['tool'];
+    expect(LCDiagramEditorCtrl.isUserDrawing()).toBe(true);
+  });
+
+  it('should save svg file created by literallyCanvas', function(done) {
+    // responseText contains a XSSI Prefix, which is represented by )]}'
+    // string. That's why double quotes is being used here. It's not
+    // possible to use \' instead of ' so the XSSI Prefix won't be
+    // evaluated correctly.
+    /* eslint-disable quotes */
+    var responseText = ")]}'\n{ \"filename\": \"imageFile1.svg\" }";
+    /* eslint-enable quotes */
+
+    LCDiagramEditorCtrl.lc.currentSvg = linesvg;
+    // @ts-ignore in order to ignore JQuery properties that should
+    // be declarated.
```

Comment:
Here and elsewhere (lots of places below), "declarated" isn't a word :)

Also adding @nishantwrp @vojtechjelinek to advise on whether this is OK.

Reply:
declarated spelling corrected. Done.

Reply:
The ts compiler expects a function exactly similar to `JQuery.ajax` here. So, I think it's fine to use `@ts-ignore` here.

/cc @vojtechjelinek @ankita240796